code/modules/client/preference/preferences.dm

```diff
@@ -2170,6 +2139,38 @@ var/global/list/special_role_times = list( //minimum age (in days) for accounts
 
 	character.change_eye_color(r_eyes, g_eyes, b_eyes)
 
+	if(disabilities & DISABILITY_FLAG_FAT && character.species.flags & CAN_BE_FAT)
+		character.dna.SetSEState(FATBLOCK,1,1)
+		character.mutations += FAT
+		character.mutations += OBESITY
+		character.overeatduration = 600
+
+	if(disabilities & DISABILITY_FLAG_NEARSIGHTED)
+		character.dna.SetSEState(GLASSESBLOCK,1,1)
+		character.disabilities |= NEARSIGHTED
+
+	if(disabilities & DISABILITY_FLAG_EPILEPTIC)
+		character.dna.SetSEState(EPILEPSYBLOCK,1,1)
+		character.disabilities |= EPILEPSY
+
+	if(disabilities & DISABILITY_FLAG_DEAF)
+		character.dna.SetSEState(DEAFBLOCK,1,1)
+		character.disabilities |= DEAF
+
+	if(disabilities & DISABILITY_FLAG_BLIND)
+		character.dna.SetSEState(BLINDBLOCK,1,1)
+		character.disabilities |= BLIND
+
+	if(disabilities & DISABILITY_FLAG_MUTE)
+		character.dna.SetSEState(MUTEBLOCK,1,1)
+		character.disabilities |= MUTE
+
+	S.handle_dna(character)
+
+	if(character.dna.dirtySE)
+		character.dna.UpdateSE()
+	domutcheck(character, null, MUTCHK_FORCED)
+
```

Comment:
Could be. I don't 100% remember my rationale behind moving this code down here but I figured this is where it'd be the safest to have it. Just in case, if nothing more